data_scrapers/san_francisco_county.py

```diff
@@ -4,101 +4,238 @@
-    """Group data by transmission category"""
-    transmission_query = '?$select=transmission_category, sum(case_count)&$group=transmission_category'
-    return get_json(transmission_url, transmission_query)
-
-# COVID+ patients from all SF hospitals in ICU vs. acute care, by date
-def get_hospitalization_json() -> Dict:
-    """Get number of COVID+ patients by bed category (ICU or regular), order by date ascending."""
-    date_order_query = '?$order=reportdate'
-    return get_json(hospitalizations_url + date_order_query)
-
-def get_icu_beds() -> Dict:
-    """group data by bed type: ICU or regular"""
-    icu_query = '?$select=dphcategory, sum(patientcount)&$group=dphcategory'
-    return get_json(hospitalizations_url + icu_query)
-
-# Daily count of tests with count and percent of positive tests
-def get_tests() -> Dict:
-    """Get tests by day, order by date ascending"""
-    date_order_query = '?$order=result_date'
-    return get_json(tests_url + date_order_query)
-
-def get_test_totals() -> Dict:
-    """Get total tests and total number of positive tests, to date"""
-    total_test_query = '?$select=sum(tests) as tests'
-    tests = get_json(tests_url + total_test_query)
-    total_positives_query = '?$select=sum(pos) as total_positives'
-    positives = get_json(tests_url + total_positives_query)
-    tests[0].update(positives[0])
-    return tests
+    response = requests.get(race_ethnicity_url)
+    response.raise_for_status()
+    # Dict of target_label : source_label for re-keying.
+    # Note: Native_Amer not currently reported
+    RACE_ETH_KEYS = {"Latinx_or_Hispanic": "Hispanic or Latino", "Asian": "Asian", "African_Amer": "Black or African American",
+                 "White": "White", "Pacific_Islander": "Native Hawaiian or Other Pacific Islander", "Native_Amer": "Native American", "Multiple_Race": "Multiple Race",
+                 "Other": "Other", "Unknown": "Unknown"}
+    data = response.json()
+    # re-key and aggregate to flatten race x ethnicity
+    race_eth_data: Dict[str,int] = { k:0 for k in RACE_ETH_KEYS.keys() } # initalize all categories to 0 for aggregating
+
+    for item in data: # iterate through all race x ethnicity objects
+        cases = int(item["confirmed_cases"])
+        cols = item.keys()
+
+        # handle unknown
+        # sum over only items for which BOTH race and ethnicity are Unknown, or one is Unknown and the other is not reported
+        if "race" in cols and "ethnicity" in cols:
+            if item["race"] == "Unknown" and item["ethnicity"] == "Unknown": # both race and ethnicity are Unknown
+                race_eth_data["Unknown"] += cases
+        elif "ethnicity" in cols and item["ethnicity"] == "Unknown": # or, race not reported and ethnicity is unknown
+            race_eth_data["Unknown"] += cases
+        elif "race" in cols and item["race"] == "Unknown": # or, ethnicity not reported and race is unknown
+            race_eth_data["Unknown"] += cases
```

Comment:
Is there a meaningful difference between “not reported” and “unknown”? It doesn’t seem like it in the logic here, so you could probably simplify and make it easier to follow with:

```py
race = item.get('race', 'Unknown')
ethnicity = item.get('ethnicity', 'Unknown')

# Now you always have a value for both, so there's no need for the complex
# branching, and no need for the `if 'race' in cols and ...` checks:
if race == 'Unknown' and ethnicity == 'Unknown':
    race_eth_data['Unknown'] += cases

if ethnicity == 'Hispanic or Latino':
    race_eth_data["Latinx_or_Hispanic"] += cases

# etc.
```